ibm/resource_ibm_container_alb_create.go

```diff
@@ -0,0 +1,198 @@
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The type of Ingress image that you want to use for your ALB deployment.",
+			},
+			"ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+				//ForceNew:    true,
+				Description: "The IP address that you want to assign to the ALB.",
+			},
+			"nlb_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The version of the network load balancer that you want to use for the ALB.",
+			},
+			"alb_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				//ForceNew:    true,
+				Description: "The type of ALB that you want to create.",
+			},
+			"vlan_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				//ForceNew:    true,
+				Description: "The VLAN ID that you want to use for your ALBs.",
+			},
+			"zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The zone where you want to deploy the ALB.",
+			},
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the cluster that the ALB belongs to.",
+			},
+
+			//response
+			"alb_id": {
+				Type: schema.TypeString,
+				//Required:    false,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The ID of the application load balancer (ALB).",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ALB name",
+			},
+			"disable_deployment": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"enable"},
+				Description:   "Set to true if ALB needs to be disabled",
+			},
+			"user_ip": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "IP assigned by the user",
+			},
+			"replicas": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "number of instances",
+			},
+			"resize": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "resize",
+			},
+		},
+	}
+}
+
+func resourceIBMContainerClassicAlbCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Println("resourceIBMContainerClassicAlbCreate")
+	albClient, err := meta.(ClientSession).ContainerAPI()
+	if err != nil {
+		return err
+	}
+	albAPI := albClient.Albs()
+
+	// "cluster":"string" //mandatory
+	// "enableByDefault": true,
+	// "ingressImage": "string",
+	// "ip": "string",
+	// "nlbVersion": "string",
+	// "type": "string", //mandatory
+	// "vlanID": "string", //mandatory
+	// "zone": "string" //mandatory
+
+	var cluster string
+	if v, ok := d.GetOkExists("cluster"); ok {
+		cluster = v.(string)
+	} else {
```

Comment:
We don't need specific else block...Terraform internally takes care since its a required variable...If user doesn't set it will fail